core/filters.py

```diff
@@ -43,18 +44,12 @@ class ProgramListFilterSet(django_filters.FilterSet):
     )
 
 
-    def __init__(self, data, *args, **kwargs):
-        data = data.copy()
+    def __init__(self, data=None, *args, **kwargs):
+        data = data.copy() if data else None
         super().__init__(data, *args, **kwargs)
 
-        colleges_choices = []
-
-        for college in College.objects.all():
-            colleges_choices.append(
-                (college.id, college.full_name.replace('College of ', ''))
-            )
-
-        self.filters['colleges'].extra['choices'] = colleges_choices
+         # Customize label display for colleges
+        self.filters['colleges'].field.label_from_instance = lambda obj: obj.full_name.replace("College of ", "") if obj.full_name.startswith("College of ") else obj.full_name
```

Comment:
Why a lambda here instead of a list comprehension? Why either instead of a traditional for loop? Does the lambda run faster?

Reply:
No, lambdas aren’t faster than a regular function.
List comprehensions ([item for item in ...]) are used to build lists. But label_from_instance doesn't want a list. it wants a function that it can call on each object individually when needed.

Reply:
We could use probably something like the following if we don't want to use lambda which is syntactic sugar. 

def get_college_label(obj):
    if obj.full_name.startswith("College of "):
        return obj.full_name.replace("College of ", "")
    return obj.full_name

self.filters['colleges'].field.label_from_instance = get_college_label

Reply:
I use the old style function.